roslibrust_codegen/src/lib.rs

```diff
@@ -0,0 +1,230 @@
+use proc_macro2::TokenStream;
+use quote::{format_ident, quote};
+use roslibrust_rospack as rospack;
+use std::collections::{BTreeMap, VecDeque};
+use std::path::PathBuf;
+use std::str::FromStr;
+use syn::parse_quote;
+
+mod parse;
+use parse::*;
+
+pub fn find_and_generate_ros_messages(additional_search_paths: Vec<PathBuf>) -> TokenStream {
+    let mut search_paths = rospack::get_search_paths();
+    search_paths.extend(additional_search_paths.into_iter());
+    let mut packages = rospack::crawl(search_paths);
+    packages.dedup_by(|a, b| a.name == b.name);
+    let mut message_files = packages
+        .iter()
+        .map(|pkg| {
+            rospack::get_message_files(pkg)
+                .unwrap()
```

Comment:
Should be an expect with a more clear error message "Failed to get message files for pkg XXX due to XXX"